proposals/0243-character-literals.md

```diff
@@ -0,0 +1,107 @@
+
+ Use of single quotes for character/scalar literals is highly precedented in other languages, including C, Objective-C, C++, Java, Elm, and Rust, although different languages have slightly differing ideas about what a “character” is. We choose to use the single quote syntax specifically because it reinforces the notion that strings and character values are different: the former is a sequence, the later is an element (though a single element can itself be a `String`). Character types also don’t support string literal interpolation and can be obtimized, which is another reason to move away from double quotes.
+ 
+ Advantages for a developer to migrate to the single quote distinction:
+ 
+  * Differentiate in the source when a literal is intended to be a Character or UnicodeScalar
+  * Distinct default type of `Character` making available that type's methods and properties.
+  * Compile time best-effort check that the literal is in fact a single Character/Unicode grapheme.
+
+ Improvements to the new implementation over that previously reviewed:
+ 
+  * SingleQuoted literals have their own new `ExpressibleBy` marker protocols preventing source breaking changes to the use of double quoted literals in existing source.
+  * Distinct protocol for ASCII literals further localising the more contentious integer conversions.
+ 
+ ### Example usage
+ 
+ Some expressions using single quoted literal syntax, their value and their type:
+
+```Swift
+	'€' // >€< Character
+	'€' as String // >€< String
+	"1"+"1" // >11< String
+	"1"+'€' // >1€< String
+	'1'+'1' as String // >11< String
+	'1'+'1' as Int // >98< Int
+	Int("0123") as Any // >Optional(123)< Optional<Int>
+	Int('€') as Any // >nil< Optional<Int>
+	Int('3') // >51< Int
+	'a'+1 //  >98< Int
+	['a', 'b'] as [Int8], // >[97, 98]< Array<Int8>
+	'a' * 'b' as Int8, // overflows at compilation
+	'b' - 'a' + 10 // >11< Int
+	"123".firstIndex(of: '2') as Any 
+		// >Optional(Swift.String.Index(_rawBits: 65799))< Optional<Index>
+	'👩🏼‍🚀'.asciiValue as Any /// >nil< Optional<UInt8>
+	('😎' as UnicodeScalar).value // >128526< UInt32
+	('👩🏼‍🚀' as UnicodeScalar).value // compilation error
+```
+ ### Single quotes in Swift, a historical perspective
+
+ In Swift 1.0, single quotes were reserved for some yet-to-be determined syntactical purpose. Since then, pretty much all of the things that might have used single quotes have already found homes in other parts of the Swift syntactical space:
+
+ - syntax for [multi-line string literals](https://github.com/apple/swift-evolution/blob/master/proposals/0168-multi-line-string-literals.md) uses triple quotes (`"""`)
+
+ - string interpolation syntax uses standard double quote syntax. 
+
+ - raw-mode string literals settled into the `#""#` syntax. 
+
+ - In current discussions around [regex literals](https://forums.swift.org/t/string-update/7398/6), most people seem to prefer slashes (`/`) or `#//#` syntax.
```

Comment:
fast forward this discussion: bare-slash regex has already landed in the toolchain :)